lib/src/core/verifier/sample/mod.rs

```diff
@@ -0,0 +1,134 @@
+use anyhow::{anyhow, Context, Result};
+use std::path::PathBuf;
+extern crate serde;
+use self::serde::{Deserialize, Serialize};
+use super::*;
+use crate::core::verifier::policy::opa::*;
+use crate::fs::*;
+use async_trait::async_trait;
+use serde_json::{json, Value};
+
+mod policy;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Tcb {
+    is_debuggable: bool,
+    cpusvn: u64,
+    svn: u64,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Quote {
+    is_debuggable: bool,
+    cpusvn: u64,
+    svn: u64,
+}
+
+#[derive(Debug)]
+pub struct Sample {
+    workdir: PathBuf,
+}
+
+impl Sample {
+    pub async fn new(dir: PathBuf) -> Result<Self> {
+        let sample = Self { workdir: dir };
+        sample.default().await?;
+        Ok(sample)
+    }
+}
+
+#[async_trait]
+impl TEEtraits for Sample {
+    fn parse(&self, evidence: &Value) -> Result<Evidence> {
+        let qstr = evidence["quote"]
+            .as_str()
+            .ok_or(anyhow!("Extract <quote> failed"))?;
+        let v: Value = serde_json::from_str(qstr)?;
+        let quote = Quote {
+            is_debuggable: v["is_debuggable"]
+                .as_bool()
+                .ok_or(anyhow!("Extract <is_debuggable> failed"))?,
+            cpusvn: v["cpusvn"]
+                .as_u64()
+                .ok_or(anyhow!("Extract <cpusvn> failed"))?,
+            svn: v["svn"].as_u64().ok_or(anyhow!("Extract <svn> failed"))?,
+        };
+        let ev = Evidence {
+            quote: serde_json::to_string(&quote)?,
+            ehd: "".to_string(),
+            add: "".to_string(),
```

Comment:
Why do this parsing manually? Can't you just deserialize the JSON directly into the evidence struct?

Reply:
Yes, use the `#[derive(Serialize, Deserialize, Debug)]` traits are better.